Add taxonomy sidebar to pages displaying the new navigation

mobaig opened the pull request:
Use the taxonomy sidebar component in static and rework the tests to
assert for the presence of the sidebar where appropriate.

Along with the taxon breadcrumbs this change is part of the education nav AB test.

## Old nav ('A' version)

![window](https://cloud.githubusercontent.com/assets/519250/23404315/33867e9e-fdac-11e6-9785-2c7bcea1c145.png)


## New nav ('B' version)

Note that this PR adds the sidebar only - the breadcrumbs are already implemented in this repo.

![window](https://cloud.githubusercontent.com/assets/519250/23404451/fe166624-fdac-11e6-84fc-9d1fa1e9fd33.png)


https://trello.com/c/0BD3POzu/403-show-parent-taxons-in-the-side-navigation-on-all-education-content-pages

nickcolley commented:
Can you update the pull request with some screenshots of how this looks?

mobaig commented:
@nickcolley yep, on it!

nickcolley commented:
Should these be inline?

![image](https://cloud.githubusercontent.com/assets/2445413/23370813/d23f28aa-fd0d-11e6-95a3-ee0623a4bc8c.png)

fofr commented:
> https://cloud.githubusercontent.com/assets/519250/23370264/11b7d4ac-fd0c-11e6-9bae-ab68f1c0cb69.png
Reckon: Content to navigation ratio feels too low.
Is this a screenshot from a prototype?

fofr commented:
Does wraith reveal any changes after 944ec04?
Order in CSS includes matters. There _may_ have been an inferred order that got lost.

eg Withdrawal notices depend on notices being included first:
```@mixin withdrawal-notice {
  .withdrawal-notice {
    @include notice;
  }
}```

alextea commented:
Just to confirm, this won't be displayed to any users yet?

This isn't the final design of the sidebar.

fofr commented:
Can we have a screenshot that shows what this PR is adding?

mobaig commented:
@nickcolley checking against how this looks in the other frontends, yes. Updated.

@fofr :

> Is this a screenshot from a prototype?

It's a screenshot from gov-frontend running in development. The tags displayed in the sidebar are made up, intended to illustrate how it looks when a piece of content is tagged to more than one level of the taxonomy. Most education content will only be tagged to one taxon.

>  There may have been an inferred order that got lost.

I've removed that commit. Was intended to be a bit of incidental tidy-up but would rather not bother if there's a chance it's broken a page render somewhere.

> Can we have a screenshot that shows what this PR is adding?

I've added before and after screenshots - does that help?

@alextea it won't, no. It's part of the existing A/B test logic that implements the taxon breadcrumb, so will only appear once we start showing the B version.

carvil commented:
@nickcolley / @fofr / @alextea This is the first step in adding the skeleton side bar for the Education navigation A/B test in this repo. The next step in this is to add related links from search via "more like this" queries, which has been worked on in `govuk_navigation_helpers` by another team member [here](https://github.com/alphagov/govuk_navigation_helpers/pull/41). 

**An important note**: all this is behind the A/B test and a feature flag, and is currently only enabled on integration. No user will see this in production until we release the full navigation. We are building this across multiple applications and are slowly getting it to its final state. More details in this comment: https://github.com/alphagov/manuals-frontend/pull/200#issuecomment-281985310

Before this goes live next week, there will definitely be a review from everyone involved. Please let us know if you would like to see more about the new navigation across all 6 affected apps and what the plans are for the release.

Here is a screenshot from the current prototype that we are aiming to implement (in terms of the side bar):

<img width="1114" alt="screen shot 2017-03-01 at 10 37 25" src="https://cloud.githubusercontent.com/assets/416701/23456384/1f2739c4-fe6b-11e6-885b-95041d6b526d.png">

When today's research round is completed (which will include testing that side bar from the prototype), we will have the final decision on what goes live and what doesn't, and will be able to tidy everything up before the release next week.

I hope this makes sense.
